appeals/web/src/server/lib/display-page-formatter.js

```diff
@@ -0,0 +1,116 @@
+import logger from '#lib/logger.js';
+import { buildHtmSpan } from '#lib/nunjucks-template-builders/tag-builders.js';
+export const formatDocumentActionLink = (
+	/** @type {any} */ appealId,
+	/** @type {any} */ listOfDocuments
+) => {
+	// If list of documents is in the correct format
+	if (
+		listOfDocuments != null &&
+		typeof listOfDocuments === 'object' &&
+		'documents' in listOfDocuments
+	) {
+		if (Array.isArray(listOfDocuments.documents) && listOfDocuments.documents.length > 0) {
+			return `/appeals-service/appeal-details/${listOfDocuments.documents[0].caseId}/documents/${listOfDocuments.documents[0].folderId}/upload/${listOfDocuments.documents[0].id}`;
+		}
+		return `/appeals-service/appeal-details/${appealId}/documents/${listOfDocuments.folderId}/upload/`;
+	}
+	return `#`;
+};
+
+export const formatListOfNotificationMethodsToHtml = (
+	/** @type {string | any[]} */ notificationMethods
+) => {
+	let html = ``;
+	if (notificationMethods.length > 0) {
+		for (const method of notificationMethods) {
+			html += `<span>${method.name}</span>`;
+			html += notificationMethods.indexOf(method) !== notificationMethods.length - 1 ? `</br>` : ``;
+		}
+	}
+	return html;
+};
+
+/**
+ * @param {string} answer
+ * @param {string} details
+ */
+export const formatAnswerAndDetails = (answer, details) => {
+	return answer === 'Yes'
+		? `${buildHtmSpan(answer)}<br>${buildHtmSpan(details)}`
+		: `${buildHtmSpan(answer)}`;
+};
+
+export const formatListOfAppeals = (
+	/** @type {{appealId: Number, appealReference: string}[]} */ listOfAppeals
+) => {
+	if (listOfAppeals.length > 0) {
+		let formattedLinks = ``;
+		for (let i = 0; i < listOfAppeals.length; i++) {
+			const appealReferenceFragments = listOfAppeals[i].appealReference.split('/');
+			const shortAppealReference = appealReferenceFragments?.[appealReferenceFragments.length - 1];
```

Comment:
Perhaps the `appealShortReference` function (`appeals/web/src/server/lib/nunjucks-filters/appeals.js`) could be used here rather than duplicating the functionality?

Reply:
Yes, we need to handle the scenario of appeal refs without slashes, which will be the case for new cases going through the BO. That function works fine with both ...